seacatauth/otp/service.py

```diff
@@ -38,141 +41,166 @@ async def _on_tick(self, event_name):
 		await self._delete_expired_totp_secrets()
 
 
-	async def get_totp_secret(self, session, credentials_id):
+	async def deactivate_totp(self, credential_id: str):
 		"""
-		Returns the status of TOTP setting.
-		If not activated, it also generates and returns a new TOTP secret.
+		Delete active TOTP secret for requested credentials.
 		"""
-		credentials = await self.CredentialsService.get(credentials_id, include=frozenset(["__totp"]))
-		secret = credentials.get("__totp")
-		if secret is not None and len(secret) > 0:
-			return {
-				"result": "OK",
-				"active": True
-			}
+		if not await self.has_activated_totp(credential_id):
+			raise TOTPNotActiveError(credential_id)
+
+		await self.StorageService.delete(collection=self.TOTPCollection, obj_id=credential_id)
```

Comment:
this raises KeyError if the object does not exist in the collection. such scenario will happen when the user wants to deactivate their TOTP stored "the old way" in the credentials collection. a simple try-except (+comment) should fix this.